dash-renderer/src/actions/index.js

```diff
@@ -917,6 +915,8 @@ function updateOutput(
                                 );
                             });
                         }
+
+                        dispatch(setAppIsReady());
```

Comment:
This is possibly the smallest change that brings the most performance benefits: only evaluate the ready state if (1) the layout was set or (2) this code here - when a `children` prop is updated

Reply:
@alexcjohnson This may not be as granular as what you had in mind but it's still a pretty significant improvement over the previous usage. I'm having a look at the surrounding code and familiarizing myself with it. For now I don't feel comfortable making a more involved change here. This can either be seen as an intermediary step (the tests are all still passing) or final.

Reply:
Tested this out against the docs `/all` page - 31 rounds of render occur with `1.4.1` over about 20 seconds with 2138 components being iterated on each prop update. Total execution time of `setAppIsReady` for these 31 executions is under 50ms. I don't see a point optimizing this further at this point even if it is possible to do so.